src/tip-opengov.ts

```diff
@@ -29,40 +31,73 @@ export async function tipOpenGov(opts: {
   if ("error" in track) {
     return { success: false, errorMessage: track.error };
   }
+  const contributorAddress = contributor.account.address;
 
   const proposalTx = api.tx.utility.batch([
     api.tx.system.remark(formatReason(tipRequest)),
-    api.tx.treasury.spend(track.value.toString(), contributor.account.address),
+    api.tx.treasury.spend(track.value.toString(), contributorAddress),
   ]);
   const encodedProposal = proposalTx.method.toHex();
   const proposalHash = blake2AsHex(encodedProposal);
+  const encodedLength = Math.ceil((encodedProposal.length - 2) / 2);
```

Comment:
What is this calculation? It needs to be the SCALE encoded length.

Reply:
before it was `proposalTx.length`, but this I took from https://github.com/polkadot-js/apps/blob/2d295e33f9d37d6582d97b9e93df81d16e1950e2/packages/page-preimages/src/Preimages/Add/Partial.tsx#L44 

@ggwpez could you please explain what is _SCALE_ encoded?

Reply:
It first does `toHex` and then uses `(x - 2) / 2` to get the number of bytes in the hex string… wtf  
Maybe there is no easier way to encode it or something. Ideal would be an `encode` or `encodedLength` function.  
Otherwise just add a test; i dont want to block over this if it works.

[SCALE](https://docs.substrate.io/reference/scale-codec/) is the encoding that Substrate/Polkadot uses for mostly everything. Especially stuff like transactions and their arguments (so also preimages).

Reply:
I'll investigate and we will add tests for sure